Src/PerformanceCollector/Perf.Shared/QuickPulseTelemetryProcessor.cs

```diff
@@ -463,7 +470,7 @@ private void ProcessTelemetry(ITelemetry telemetry)
                             documentStreams,
                             documentStream => documentStream.RequestQuotaTracker,
                             documentStream => documentStream.CheckFilters(telemetryAsRequest, out groupErrors),
-                            ConvertRequestToTelemetryDocument);
+                            requestTelemetry => ConvertRequestToTelemetryDocument(requestTelemetry, this.config?.ApplicationIdProvider));
```

Comment:
let's not mutate the original request telemetry. The logic with post-processing by itself already hard to understand. It is harder and more error-prone when opening live stream UI may produce different results than when it's not opened.

Reply:
For this reply: 
![image](https://user-images.githubusercontent.com/9950081/57101025-34222e80-6cd5-11e9-99e3-2058218e31ee.png)

I only suggest to do it for the url population. There will be no copy involved

Reply:
Makes sense. It appears that Source field is not used by QP after all, also removed its' evaluation here.